### PR TITLE
Add Gatsby Cloud-related webhooks to Enhanced Autobuilder

### PR DESCRIPTION
**Pull request by rayangler**
### Ticket

DOP-3968

### Notes

* Adds existing v1 webhooks for Gatsby Cloud builds to the enhanced Autobuilder. The logic should be the same for v1 and v2 as they are fully exported from their respective `api/handlers/*.ts` files.
* Envs needed for such webhooks should have already been added in previous PRs, but please let me know if I missed something! Notably, the envs include the following and can be found defined in [default.json](https://github.com/mongodb/docs-worker-pool/blob/ac389132d575eeb8f0035db7d6330f974f611fd8/cdk-infra/static/api/config/default.json), [custom-environment-variables.json](https://github.com/mongodb/docs-worker-pool/blob/ac389132d575eeb8f0035db7d6330f974f611fd8/cdk-infra/static/api/config/custom-environment-variables.json), and [ssm.ts](https://github.com/mongodb/docs-worker-pool/blob/ac389132d575eeb8f0035db7d6330f974f611fd8/cdk-infra/utils/ssm.ts):
  * Snooty secret
  * GitHub bot password
  * GitHub deletion ending

### CODE REVIEW

**Review by mmeigs — COMMENTED** [review]
To my eye it looks good! imo @branberry should approve this before merge because of his knowledge-base in this new construct.

**Review by branberry — APPROVED** [review]
LGTM! I added one comment, but it's tangential to the changes you've made and doesn't block you from merging this PR.

File: cdk-infra/lib/constructs/api/webhook-api-construct.ts
Comment (rayangler) [reply cdk-infra/lib/constructs/api/webhook-api-construct.ts:79]: That makes sense, and I agree. I think we should make it a separate task to avoid the risk of introducing any potential regression so soon to UAT. I can create a ticket for something like this later

**Pull request merged**
